### ISSUE TITLE
Detection accuracy metrics are lower than those of other Repositories？

### ISSUE THREAD

**Issue by zhuyy116** [ultralytics/ultralytics#19397]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

## Environment Information

- **Operating System**: Ubuntu 22.04
- **Python Version**: 3.8
- **PyTorch Version**: 1.12
- **GPU**: 8 GPUs

## Issue Description

I am experiencing lower mAP values when using the Ultralytics repository compared to the [YOLOv10](https://github.com/THU-MIG/yolov10) repository. Here are the specific details:

### Models and Datasets

| Model       | Dataset       | Ultralytics mAP | YOLOv10 mAP | [MMYOLO](https://github.com/open-mmlab/mmyolo) mAP |
|-------------|---------------|-----------------|-------------|-------------|
| YOLOv10s    | VisDrone      | 0.171           | 0.221       | - |
| YOLOv10s    | [RUOD](https://github.com/xiaoDetection/RUOD) | 0.556           | 0.622       | - |
| YOLOv8s     | RUOD  |     0.558     | 0.615       | 0.606 |

### Reproduction Code

```python
from ultralytics import YOLO
from ultralytics.utils import SettingsManager

model = YOLO('ultralytics/cfg/models/v10/yolov10s.yaml', task='detect')

settings = SettingsManager()
settings.update(tensorboard=True,
                wandb=False,)

model.train(data='ultralytics/cfg/datasets/VisDrone.yaml',
            epochs=300,
            patience=0,
            batch=16 * 8,
            imgsz=640,
            save=True,
            save_period=-1,
            cache=True,
            device=[6,1,2,3,4,5,7,8],
            workers=8,
            project='outputs/yolov10s_VisDrone',
            name='origin',)
```

## Additional

I have attempted to adjust the Ultralytics code based on YOLOv10 modifications, but it did not yield the expected results. The outputs _args.yaml_ did not show significant differences. I would like to understand the reasons behind these discrepancies and whether it is possible to reproduce the YOLOv10 repository results using the Ultralytics repository.
Thank you for your assistance in resolving this issue.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zhuyy116, thank you for bringing up this question and engaging with the Ultralytics community 🚀! We encourage you to explore our [Docs](https://docs.ultralytics.com/) for helpful resources, including [Python usage](https://docs.ultralytics.com/usage/python/) and [CLI usage](https://docs.ultralytics.com/usage/cli/) examples. These may provide clarity on some commonly encountered scenarios.

To address your concern:

If this is a 🐛 Bug Report, please share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) to help us accurately debug the issue. This should include the exact code, configurations, and datasets you're using for your experiments.

If this is a ❓ Question about training or reproducibility, please provide:
- A precise description of any code modifications or hyperparameter changes.
- Example images or samples from your datasets.
- Logs or metrics from your training runs.

Be sure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), particularly in areas such as data preparation and augmentation strategies.

## Upgrade

We recommend ensuring that you are running the latest version of `ultralytics` along with all its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Upgrade to the newest release in a [**Python>=3.8**](https://www.python.org/) environment with a properly installed [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

This guarantees that any enhancements or bug fixes in the latest version are included in your workflow.

## Verified Environments

Our YOLO models are tested and ready to run in the following verified environments. You can explore various platforms optimized for performance and reproducibility:
- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon Web Services** Deep Learning AMI: Refer to [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Learn to use our robust Docker image from the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

Have complex questions or need peer support? 🎧 Chat in real-time with the community on [Discord](https://discord.com/invite/ultralytics), explore thoughtful discussions on [Discourse](https://community.ultralytics.com/), or exchange information on the [Ultralytics Subreddit](https://reddit.com/r/Ultralytics).

## Continuous Integration Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
The badge above shows our CI test status. Green indicates full operational correctness of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response 🤖, but rest assured that an Ultralytics engineer will review your question and assist further shortly. Thank you for your patience and understanding! 😊

**Comment by Y-T-G** [comment]
Probably because you're using cache which lowers mAP.

**Comment by zhuyy116** [comment]
> Probably because you're using RAM cache which lowers mAP.
> 
> You can try disk cache instead `cache="disk"`

But the Ultralytics and YOLOv10 repositories are the same setup, and the model mAP will be higher in YOLOv10 than in Ultralytics.

**Comment by Y-T-G** [comment]
The YOLOv10 fork is using an Ultralytics version where RAM cache is broken. It was fixed in a later version.

**Comment by Y-T-G** [comment]
https://github.com/ultralytics/ultralytics/pull/9828

**Comment by zhuyy116** [comment]
> The YOLOv10 fork is using an Ultralytics version where RAM cache is broken. It was fixed in a later version.

I tried the setting` cache=“disk”` and the metrics are the same for both repository models. Thank you very much.

**Comment by glenn-jocher** [comment]
Thank you for following up. The observed performance differences between implementations could stem from subtle architectural or training configuration variations in the original YOLOv10 repository. For optimal reproduction of their results, we recommend: 

1. Verifying your training matches their exact hyperparameters and preprocessing from the [YOLOv10 documentation](https://docs.ultralytics.com/models/yolov10/)  
2. Ensuring you're using matching Ultralytics/YOLOv10 repo versions

We're happy to investigate further if you can share:
- Training logs from both implementations
- Exact package versions used (`pip list` outputs)

Let us know how we can assist! 🚀
